### DISCUSSION TITLE
discuss: bilibili: default "open" behavior for multi part video

### DISCUSSION THREAD

**Discussion by rhjdvsgsgks** [InfinityLoop1308/PipePipe#131] [Ideas]
for now, the behavior of open a multi part video is open the first part in the multi part video. and show the playlist as second item in related tab. i think it would be great to open the playlist by default (no matter open from link, related, or a playlist), rather than the first part of a multi part video. in another words: treat multi part video as a playlist directly. not sure does newpipe support show a playlist (multi part video) as a item in a play list.

(sorry for make too much request. i cant get access to a computer right now. therefore unable to debug it. if you are not interested in this suggestion. you can keep it open until i got a computer and make a pr in september)

**Comment by InfinityLoop1308** [comment]
I don't think it is possible under the current framework without writing patches, and even if possible, there would still be problems. For example, the currently playing partition would be added to `History`, which would be strange if you were to add the playlist instead.

**Comment by InfinityLoop1308** [comment]
Also you will have to manually update the all the extracting logics where it could appear, including searching, related items, etc.

**Comment by rhjdvsgsgks** [comment]
> For example, the currently playing partition would be added to `History`, which would be strange if you were to add the playlist instead.

but user is playing a part of video. not the playlist itself. so i think we should still add the video (a part of multi part video) to history, rather than a playlist (multi part video)

and. you are right. change logic to this will make a big refractor to current extractor logic. in my understanding:

current logic:
bvxxxx is a video (with `p` default to 1 , unless specific the p to other number). this is unchanged if video is multi part. and show a extra playlist in 2nd item in related, as index of multi part video 

logic discussed in this issus:
bvxxxx is possible to a video (single part video) or a playlist (multi part video). when video is multi part, `bvxxxx?p=n` is a video. otherwise, `bvxxxx` is a video

seems like lot of work need to do

**Comment by rhjdvsgsgks** [comment]
i just found that it will also have issue if treat multi part video as playlist. because a channel can update their video and add a new part to it. if a video is single part when user added to local playlist, and then become a multi part video. in this case playlist are nested. which will cause trouble

**Comment by rhjdvsgsgks** [comment]
so i think your solution is the best so far. but it have a little bit buggy.
when playing a multi part video. next part will be added to queue when current part started. thats what we expected. but when network reconnected and video reload, next part willl also added to the queue. which caused duplicate

**Reply by InfinityLoop1308** [reply]
you may want to open an issue.

**Comment by rhjdvsgsgks** [comment]
and i have some suggestion is that:
add all parts to the queue when any part of multi p video been played (just like the "play all" button in the playlist, the only difference is that play the user selected part rather than play from first part). it much more convenience than only add next part to the queue.  because user can jump between different p without exit fullscreen and go to playlist by using the queue menu
